pandas/core/indexing.py

```diff
@@ -1091,8 +1095,17 @@ def _convert_to_indexer(self, obj, axis=0, is_setter=False):
         """
         labels = self.obj._get_axis(axis)
 
-        # if we are a scalar indexer and not type correct raise
-        obj = self._convert_scalar_indexer(obj, axis)
+        if isinstance(obj, slice):
+            return self._convert_slice_indexer(obj, axis)
+
+        # try to find out correct indexer, if not type correct raise
+        try:
+            obj = self._convert_scalar_indexer(obj, axis)
+        except TypeError:
+
+            # but we will allow setting
+            if is_setter:
```

Comment:
So I fixed everything, but had to change this:

What kind of index should this be:
``Index([1., 2., 3.], dtype=int)``
``Index([1., 2, 3.], dtype=int)``

w/o actually running it?